### ISSUE TITLE
Function app erroneously changes WEBSITE_CONTENTSHARE after apply

### ISSUE THREAD

**Issue by piyaldas** [hashicorp/terraform-provider-azurerm#18388]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

">= 0.12.6"

### AzureRM Provider Version

"~> 3.14.0"

### Affected Resource(s)/Data Source(s)

azurerm_windows_function_app, azurerm_windows_function_app_slot

### Terraform Configuration Files

```hcl
locals {
  fapp_app_settings = merge(var.app_settings, {WEBSITE_CONTENTSHARE = "${var.app_settings.WEBSITE_CONTENTSHARE}-a"})
  fapp_slot_app_settings = merge(var.app_settings, {WEBSITE_CONTENTSHARE = "${var.app_settings.WEBSITE_CONTENTSHARE}-b"})
}

resource "azurerm_windows_function_app" "fun_app" {
  depends_on           = [module.fapp_file_share_01]
  name                       = var.name
  location                   = var.location
  resource_group_name        = var.resource_group_name
  service_plan_id        = var.app_service_plan_id
  storage_account_name       = var.storage_account_name
  storage_account_access_key = var.storage_account_access_key

  functions_extension_version  = var.fapp_default_setting.functions_extension_version
  builtin_logging_enabled = false
  site_config {
    application_insights_connection_string = var.application_insights.application_insights_connection_string
    application_insights_key = var.application_insights.application_insights_key
    application_stack {
      dotnet_version = var.fapp_default_setting.site_config.application_stack.dotnet_version
    }
    always_on = var.fapp_default_setting.site_config.always_on
    ftps_state = var.fapp_default_setting.site_config.ftps_state
    vnet_route_all_enabled  = var.enable_vnet_integration ? true : false
  }
  https_only = var.fapp_default_setting.https_only
  app_settings               = local.fapp_app_settings
  identity {
    type = "SystemAssigned"
  }
  lifecycle {
    ignore_changes = [
      app_settings["FUNCTIONS_WORKER_RUNTIME"],
      app_settings["WEBSITE_RUN_FROM_PACKAGE"],
      app_settings["WEBSITE_CONTENTSHARE"],
      enabled,
      sticky_settings,
    ]
  }
}


resource "azurerm_windows_function_app_slot" "fun_app" {
  depends_on  = [azurerm_windows_function_app.fun_app]
  name                 = lookup(var.slot_setting, "slot_name", "staging01")
  function_app_id      = azurerm_windows_function_app.fun_app.id
  storage_account_name = var.storage_account_name
  storage_account_access_key = var.storage_account_access_key

  functions_extension_version  = var.fapp_default_setting.functions_extension_version
  builtin_logging_enabled = false
  site_config {
    application_insights_connection_string = var.application_insights.application_insights_connection_string
    application_insights_key = var.application_insights.application_insights_key
    application_stack {
      dotnet_version = var.fapp_default_setting.site_config.application_stack.dotnet_version
    }
    always_on = var.fapp_default_setting.site_config.always_on
    ftps_state = var.fapp_default_setting.site_config.ftps_state
    vnet_route_all_enabled  = var.enable_vnet_integration ? true : false
  }
  https_only = var.fapp_default_setting.https_only
  app_settings               = local.fapp_slot_app_settings
  identity {
    type = "SystemAssigned"
  }
  lifecycle {
    ignore_changes = [
      app_settings["FUNCTIONS_WORKER_RUNTIME"],
      app_settings["WEBSITE_RUN_FROM_PACKAGE"],
      app_settings["WEBSITE_CONTENTSHARE"],
      enabled,
    ]
  }
}
```

### Debug Output/Panic Output

```shell
I have updated the DUMMY app settings from 1 to 2 and this is the only change I see running the plan command:

Production Plan:
  ....
  ~ app_settings                   = {
         
          ~ "DUMMY"                                 = "1" -> "2"
            "WEBSITE_CONTENTOVERVNET"               = "1"
            "WEBSITE_CONTENTSHARE"                  = "fun-app-01-a"
        }
....        

Staging Plan:
  ....
  ~ app_settings                   = {
         
          ~ "DUMMY"                                 = "1" -> "2"
            "WEBSITE_CONTENTOVERVNET"               = "1"
            "WEBSITE_CONTENTSHARE"                  = "fun-app-01-b"
        }
....
```

### Expected Behaviour

I am expecting to go in azure portal and see below:
Function App -> App Setting -> **WEBSITE_CONTENTSHARE** = ${var.app_settings.WEBSITE_CONTENTSHARE}-**a**
Function App -> Deployment Slot -> App Setting -> **WEBSITE_CONTENTSHARE** = ${var.app_settings.WEBSITE_CONTENTSHARE}-**b**

### Actual Behaviour

The app settings is correctly set for other parameters while the Deployment Slots **WEBSITE_CONTENTSHARE** **erroneously** changed from ${var.app_settings.WEBSITE_CONTENTSHARE}-**b** to ${var.app_settings.WEBSITE_CONTENTSHARE}-**a** though within the plan there is no evidence that setting is going to change.
This is very dangerous because it can cause an **"Hidden swap"** of the running software from a stable version to an unstable one.

### Steps to Reproduce

2. Edit your terraform configuration file: for instance add a parameter in the app settings DUMMY
3. Check the plan on the standard output: the only change highlighted is the value of the DUMMY app settings. Nothing related to **WEBSITE_CONTENTSHARE**
4. Deploy on the "production" & "staging" slot

As soon as the apply is completed you will find out,
Production  **WEBSITE_CONTENTSHARE** is ${var.app_settings.WEBSITE_CONTENTSHARE}-**a** 
While Staging **WEBSITE_CONTENTSHARE** is also  **erroneously** updated to ${var.app_settings.WEBSITE_CONTENTSHARE}-**a** which should be ${var.app_settings.WEBSITE_CONTENTSHARE}-**b**

### References

I have gone through few previous TF issues reported & found below ones; I think these are re-appeared.

https://github.com/hashicorp/terraform-provider-azurerm/pull/13349
https://github.com/hashicorp/terraform-provider-azurerm/issues/10499

**Comment by xiaxyi** [comment]
@piyaldas Thanks for raising this issue. From your config, I can tell you have the storage account restricted to a vNet as`WEBSITE_CONTENTOVERVNET` sets to 1, right?  The contentshare won't be created automatically if this setting is enabled, you can refer to my [pr ](https://github.com/hashicorp/terraform-provider-azurerm/pull/18258)for more information.

**Comment by xiaxyi** [comment]
@piyaldas May I know if you still seeing this issue? The `WEBSITE_CONTENTSHARE` for slot instance and function production instance is correctly set from my side as below:
 config:
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/92154856/001d5bf3-09dc-428e-a06b-681ec604b638)

result:
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/92154856/e4c8f81a-da86-4e60-9a7d-fa511f01415d)

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/92154856/6f661ca5-b2f2-4983-9ed9-4185d6270c39)

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #18258. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!
